input/kafkamdm/lag_monitor.go

```diff
@@ -47,43 +48,107 @@ func (l *lagLogger) Min() int {
 	return min
 }
 
+type rateLogger struct {
+	sync.Mutex
+	lastOffset int64
+	lastTs     time.Time
+	rate       int64
+}
+
+func (o *rateLogger) Store(offset int64, ts time.Time) {
+	o.Lock()
+	defer o.Unlock()
+	if o.lastTs.IsZero() {
+		// first measurement
+		o.lastOffset = offset
+		o.lastTs = ts
+		return
+	}
+	metrics := offset - o.lastOffset
+
+	o.lastOffset = offset
+
+	duration := int64(ts.Sub(o.lastTs) / time.Second)
+	o.lastTs = ts
+	if duration <= 0 {
+		// current ts is > last ts. This would only happen if our clock
+		// suddenly changes, in which case we cant reliably work out how
+		// low it has really been since we last took a measurement.
+		return
+	}
+	rate := metrics / duration
```

Comment:
why would it run close to the second boundary? it's a standard ticker so it depends on when it was started.

see:
```
~/t/testticker ❯❯❯ cat main.go
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println("start time", time.Now())
	ticker := time.NewTicker(2 * time.Second)
	for ts := range ticker.C {
		fmt.Println(ts)
	}
}
~/t/testticker ❯❯❯ go run main.go
start time 2017-02-23 10:39:45.746803375 +0100 CET
2017-02-23 10:39:47.747025261 +0100 CET
2017-02-23 10:39:49.747043167 +0100 CET
^Csignal: interrupt
~/t/testticker ❯❯❯ go run main.go                                                            ⏎
start time 2017-02-23 10:39:53.88361158 +0100 CET
2017-02-23 10:39:55.883752266 +0100 CET
2017-02-23 10:39:57.883791314 +0100 CET
^Csignal: interrupt
```

> That would require using a float64 for the rate instead of an int.
￼no, we just round to an int after doing the rate calculation. i created PR #546 which shows what i mean.